docs/minio-job.md

```diff
@@ -0,0 +1,208 @@
+spec:
+  mcImage: quay.io/minio/mc:latest # optional, defaults to minio/mc:latest
+  serviceAccountName: mc-job-sa
+  securityContext: {}
+  containerSecurityContext: {}
+  tenant:
+    name: mytest-minio
+  commands:
+    - op: make-bucket
+      args:
+        name: memes
+    - name: add-my-user-1
+      op: admin/user/add
+      args:
+        user: ${USER}
+        password: ${PASSWORD}
+      envFrom:
+        - secretRef:
+            name: mytestsecretenvs
+    - name: add-my-user-2
+      op: admin/user/add
+      args:
+        user: pedro
+        password: $PASSWORD
+      env:
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mytestsecret
+              key: PASSWORD
+    - name: add-my-policy
+      op: admin/policy/create
+      args:
+        name: memes-access
+        policy: /temp/policy.json
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
+The `mcImage` field specifies the Docker image that will be used to run the mc commands. In this case, the image is `minio/mc:latest`.
+## serviceAccountName
+The `serviceAccountName` field specifies the name of the Kubernetes ServiceAccount that will be used to run the mc commands. In this case, the ServiceAccount is `mc-job-sa`.
+## securityContext
+example:
+```yaml
+runAsUser: 1000
+runAsGroup: 1000
+fsGroup: 1000
+fsGroupChangePolicy: "OnRootMismatch"
+runAsNonRoot: true
+allowPrivilegeEscalation: false
+capabilities:
+  drop:
+    - ALL
+```
+The `securityContext` field specifies the security context that will be used to run the mc commands. 
+## containerSecurityContext
+The `containerSecurityContext` field specifies the security context that will be used to run the `mc` commands in the container.
+## tenant
+```yaml
+name: tenantName
+namespace: tenantNamespace
+```
+The target tenant that the job will run against.
+## commands
+### args
+if you set this field, the `mc` command will be executed with the arguments.
+`op` must be one of these:
+`make-bucket`, `admin/user/add`,`admin/policy/create`,`admin/policy/attach`, `admin/config/set`, `support/callhome`,`license/register`
+```yaml
+op: make-bucket
+args:
+  name: memes
```

Comment:
Full list of args? No `--with-lock` analog for enabling bucket WORM locking? We can only do that from bucket creation.

Reply:
Not sure if `--with-lock: ""` is trivial for users. If we do parse the arguments, I think an argument like `with-locks: <boolean>` would be more suitable. The only drawback is that we may end up supporting all arguments and that's hard to maintain, documents, ...

We may also choose to only provide basic functionality and use `command` for more advanced scenarios. Of course, there will be a grey area what is considered basic functionality :smile:.

Reply:
This might be necessary @ramondeklein - you cannot enabling worm locking after creating a bucket at this time. S3 now supports it, but I think MinIO needs an update to make it happen. That also means older MinIO tenants wont support it either.

Reply:
@ravindk89 If `with-lock` is a common used feature, then I would propose to add it as `with-lock: true` in the job specificaition instead of `--with-lock: ""`. If you need something that we don't support as a built-in command, then you can always use:
```yaml
- op: command
  command:
  - "mc"
  - "mb"
  - "--unsupported-option"
  - "my-amazing-bucket"
```

Reply:
I don't think I understand your question. What I meant is that we should support common-used options in a user-friendly way and all "advanced" functionality can be done via the `command` syntax. I don't think that `--with-locks: ""` is a user-friendly way is you just mean to add the `--with-locks` option. I think the built-in commands should use normal identifiers for the supported arguments and not use arguments starting with `--`. If an option is enabled, then using `true` would make more sense instead of `""` (IMHO).

PS: I think if I had to use this, I would always use the `command` syntax. If you know `mc`, then you know how to use the MinIO job in `command` syntax and you don't have any surprises. I don't think I would use the "built-in" commands, but this is highly opinionated.